node/src/components/deploy_buffer.rs

```diff
@@ -146,6 +143,12 @@ impl DeployBuffer {
                 debug!("DeployBuffer: missing block at height: {}", contiguous_next);
                 return false;
             }
+
+            // if we've reached genesis via an unbroken sequence, we're good
+            // if we've seen a full ttl period of blocks, we're good
+            if *height == 0 || *timestamp < earliest_needed {
+                return true;
+            }
```

Comment:
curious why you pulled this lower?

Reply:
Because it would skip the unbroken sequence check if we reach the genesis block or the right timestamp. For example if we had blocks [0, 2, 3, 4, 5] it would still return true even though block 1 was missing.